apps/dashboard/esbuild.config.js

```diff
@@ -37,42 +37,14 @@ const prepPlugin = {
   },
 }
 
-// We can run into conflicts if we use the minified
-// version of some dependencies. So this plugin ensures
-// that we compile the source code of a given dependency
-// instead of the minified version.
-//
-// See https://github.com/OSC/ondemand/issues/3688 for more information.
-const minifiedSrcResolvePlugin = {
-  name: 'minifiedSrcResolvePlugin',
-  setup(build) {
-
-    build.onResolve({ filter: /preact|exifr.*/ }, args => {
-
-      const preactBase = `${__dirname}/node_modules/preact`;
-      const lookup = {
-        'preact': `${preactBase}/src/index.js`,
-        'preact/hooks': `${preactBase}/hooks/src/index.js`,
-        'exifr/dist/mini.esm.mjs': `${__dirname}/node_modules/exifr/src/bundles/mini.mjs`,
-      }
-
-      for (const [key, value] of Object.entries(lookup)) {
-        if(args.path == key) {
-          return { path: value }
-        }
-      }
-    })
-  },
-}
-
 esbuild.build({
   entryPoints: entryPoints,
   bundle: true,
   sourcemap: true,
   format: 'esm',
   outdir: buildDir,
   external: ['fs'],
-  plugins: [prepPlugin, minifiedSrcResolvePlugin],
+  plugins: [prepPlugin],
```

Comment:
Using this plugin was what was causing the issues in the Uppy updates. An update in the preact library cascaded into Uppy where Uppy was not really using that library correctly because some parts of that library were going through this plugin.

Removing it solves the issues at least for now. It seems if we run into #3688 again we'll have to find some other way to resolve it.